src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,130 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] objects;
+    private int size;
+
+    public ArrayList() {
+        objects = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= objects.length) {
+            this.grow();
+        }
+        objects[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size >= objects.length) {
+            this.grow();
+        }
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("index: " + index + "size: " + size);
+        }
+        if (index <= size) {
+            Object[] newObjectsArray = new Object[objects.length + 1];
+            System.arraycopy(objects, 0, newObjectsArray, 0, index);
+            newObjectsArray[index] = value;
+            System.arraycopy(objects, index, newObjectsArray, index + 1, objects.length - index);
+            objects = newObjectsArray;
```

Comment:
could we use `System.arraycopy` only ones here?

Reply:
i haven't idea how to realize it....